SeQuant/domain/mbpt/models/cc.cpp

```diff
@@ -148,4 +149,100 @@ std::vector<ExprPtr> CC::λ(bool screen, bool use_topology,
   return result;
 }
 
+std::vector<sequant::ExprPtr> CC::t_pt(std::size_t o, std::size_t r) {
+  assert(o == 1 &&
+         "sequant::mbpt::sr::CC::t_pt(): only first-order perturbation is "
+         "supported now");
+  assert(r == 1 &&
+         "sequant::mbpt::sr::CC::t_pt(): only one-body perturbation "
+         "operator is supported now");
+
+  // construct mu_bar
+  const auto mu_truncate_at =
+      r == 1 ? 2
+             : 4;  // truncate mu_bar at rank 2 for one-body perturbation
+                   // operator and at rank 4 for two-body perturbation operator
+  auto mu_bar = sim_tr(op::mu(r), mu_truncate_at);
+
+  // construct [hbar, T(1)]
+  auto hbar_pert = sim_tr(op::H(), 3) * op::pertT1(N);
+
+  // [Eq. 34, WIREs Comput Mol Sci. 2019; 9:e1406]
+  auto expr = simplify(mu_bar + hbar_pert);
+
+  // connectivity:
+  // connect t and t1 with {h,f,g}
+  // connect mu with t
+  std::vector<std::pair<std::wstring, std::wstring>> op_connect = {
+      {optype2label.at(OpType::h), optype2label.at(OpType::t)},
+      {optype2label.at(OpType::f), optype2label.at(OpType::t)},
+      {optype2label.at(OpType::g), optype2label.at(OpType::t)},
+      {optype2label.at(OpType::h), optype2label.at(OpType::t_1)},
+      {optype2label.at(OpType::f), optype2label.at(OpType::t_1)},
+      {optype2label.at(OpType::g), optype2label.at(OpType::t_1)},
+      {optype2label.at(OpType::μ), optype2label.at(OpType::t)}};
+
+  std::vector<ExprPtr> result(P + 1);
+  for (auto p = P; p >= PMIN; --p) {
+    auto freq_term = ex<Variable>(L"ω") * op::P(p) * op::pertT1_(p);
+    result.at(p) =
+        op::vac_av(op::P(p) * expr, op_connect) - op::vac_av(freq_term);
+  }
+  return result;
+}
+
+std::vector<ExprPtr> CC::λ_pt(size_t o, size_t r) {
+  assert(o == 1 &&
+         "sequant::mbpt::sr::CC::λ_pt(): only first-order perturbation is "
+         "supported now");
+  assert(r == 1 &&
+         "sequant::mbpt::sr::CC::λ_pt(): only one-body perturbation "
+         "operator is supported now");
+  // construct hbar
+  auto hbar = sim_tr(op::H(), 4);
+  // construct mu_bar
+  const auto mu_truncate_at =
+      r == 1 ? 2
+             : 4;  // truncate mu_bar at rank 2 for one-body perturbation
+                   // operator and at rank 4 for two-body perturbation operator
+  auto mu_bar = sim_tr(op::mu(r), mu_truncate_at);
+  // construct [hbar, T(1)]
+  auto hbar_pert = sim_tr(op::H(), 3) * op::pertT1(N);
+
+  // [Eq. 35, WIREs Comput Mol Sci. 2019; 9:e1406]
+  const auto One = ex<Constant>(1);
+  auto expr = simplify((One + op::Lambda(N)) * (mu_bar + hbar_pert) +
+                       op::pertLambda1(N) * hbar);
+
+  // connectivity:
+  // t and t1 with {h,f,g}
+  // projectors with {h,f,g}
+  // mu with t
+  // mu with projectors
+  std::vector<std::pair<std::wstring, std::wstring>> op_connect = {
```

Comment:
this is getting way long, need a more compact way to assemble this

Reply:
Addressed in https://github.com/ValeevGroup/SeQuant/pull/141/commits/11b05dfb32f3a33b9050b8ef617fdbb735a49d46